server/rest/src/main/java/org/infinispan/rest/resources/CounterResource.java

```diff
@@ -174,8 +192,15 @@ public Invocations getInvocations() {
       Long delta = checkForNumericParam("delta", request, responseBuilder);
       if (delta == null) return completedFuture(responseBuilder.build());
 
-      CounterConfiguration configuration = counterManager.getConfiguration(counterName);
-      if (configuration == null) return notFoundResponseFuture();
+      NonRecursiveEventLoopGroup.reserveCurrentThread();
+      CounterConfiguration configuration;
+      try {
+         configuration = counterManager.getConfiguration(counterName);
```

Comment:
you have `getConfigurationAsync(String)` if you want :) you would have to change this method :(

edit: same below :)

Reply:
That would be preferable, we don't want to use `NonRecursiveEventLoopGroup.reserveCurrentThread` if we don't need to. It is more of a bandaid until the underlying issue can be solved.

Reply:
Ah, I didn't see it because it wasn't on the interface :)